src/ExactOnline.Client.Sdk/Controllers/EntityController.cs

```
@@ -93,6 +93,12 @@ public Boolean Update(object entity)
 			var converter = new EntityConverter();
 			string json = converter.ConvertObjectToJson(OriginalEntity, entity, _entityControllerDelegate);
 
+            if (string.IsNullOrEmpty(json))
+            {
+                //Nothing to update
+                return true;

```

This is indeed a nice check. I notice that, as a result of this change, the unit test "EntityController_Update_WithNoFieldsAltered_Succeeds" is now failing. Please also update the test because now the controllerMock.Data is null instead of empty string.

---

I have updated the test an added the comment. I also added an overload to control the showing of the login form. I our application we cannot just show a dialog, so we implemented the authorization flow manually and if the authorization fails the client should just throw an exception instead of showing a dialog.